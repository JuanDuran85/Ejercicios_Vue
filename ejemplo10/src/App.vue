<template>
  <div id="app">
    <div class="titulo">
      <h1>Random Git Cat</h1>
    </div>
    <div class="formulario">
      <label for="">
        Titulo: <input type="text" v-model="titulo">
      </label>
      <label for="filtro">Filtro
        <select name="filtro" id="filtro" v-model="filtro">
          <option value="blur">blur</option>
          <option value="mono">mono</option>
          <option value="sepia">sepia</option>
          <option value="negative">negative</option>
          <option value="paint">paint</option>
          <option value="pixel">pixel</option>
        </select>
      </label>
      <label for="color">Color
        <select name="color" id="color" v-model="color">
          <option value="red">Rojo</option>
          <option value="blue">Azul</option>
          <option value="green">Verde</option>
          <option value="yellow">Amarillo</option>
          <option value="white">Blanco</option>
        </select>
      </label>
      <label for="">
          Tama√±o<input type="number" v-model="medida">
      </label>
    </div>
    <div class="muestra">
      <button @click.prevent="buscar">Mostrar Gato</button>
      <componente-imagen :src="buscarURL"/>
    </div>
  </div>
</template>

<script>
import CompoImg from './components/CompoImg.vue'

export default {
  name: 'App',
  data() {
    return {
      titulo: "Escribe el nombre",
      color: "blue",
      medida: "300",
      filtro: "blur",
      resultado: [],
    }
  },
  methods: {
    buscar(){
      console.log("entro");
      fetch(`https://cataas.com/cat/gif/says/${this.titulo}?filter=${this.filtro}&color=${this.color}&size=${this.medida}`)
      .then(respuesta => {
          this.resultado = respuesta;
        })
      .catch(error => console.error(error));
    }
  },
  components: {
    "componente-imagen": CompoImg 
  },
  computed: {
    buscarURL(){
      return this.resultado.url;
    }
  },
}
</script>

<style scope>
  .titulo {
    margin-top: -25px;
    font-size: 24px;
    font-weight: 800;
    text-align: center;
    background: rgba(136, 162, 216, 0.795);
    height: 20vh;
  }
  .titulo h1 {
    padding-top: 35px;
  }
  .formulario {
    display: block;
    padding: 5px;
    background: rgba(218, 79, 79, 0.795);
    font-size: 18px;
    text-align: center;
  }
  .formulario label {
    display: block;
    margin: 10px;
    padding: 10px;
  }
  .muestra {
    text-align: center;
    padding: 10px;
    background: rgba(136, 162, 216, 0.795);
    height: auto;
  }
</style>
