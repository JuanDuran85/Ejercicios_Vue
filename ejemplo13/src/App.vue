<template>
  <div id="app">
    <h1>Utilizando Vue-Router con Vue-Cli</h1>
    <nav>
      <ul>
        <li><router-link to="/">Inicio</router-link></li>
        <li><router-link to="/servicios">Servicios</router-link></li>
        <li><router-link to="/blog/1">Blog</router-link></li>
        <li><router-link to="/demo">Demo</router-link></li>
        <li><router-link to="/demo2">Demo2</router-link></li>
        <li><router-link :to="{ name: 'contacto' }">Contacto</router-link></li>
      </ul>
    </nav>
    <router-view></router-view>
  </div>
</template>

<script>

export default {
  name: 'App',
}
</script>

<style>
  body {
    text-align: center;
    font-family: 'Times New Roman', Times, serif;
    font-size: 20px;
    font-weight: 600;
    text-decoration: none;
    list-style: none;
  }
</style>
