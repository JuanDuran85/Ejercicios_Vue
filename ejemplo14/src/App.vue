<template>
  <div>
    <nav class="navbar navbar-light navbar-expand-lg fixed-top" id="mainNav">
        <div class="container"><a class="navbar-brand">V8</a><button data-toggle="collapse" data-target="#navbarResponsive" class="navbar-toggler" aria-controls="navbarResponsive" aria-expanded="false" aria-label="Toggle navigation"><i class="fa fa-bars"></i></button>
            <div
                class="collapse navbar-collapse" id="navbarResponsive">
                <ul class="nav navbar-nav ml-auto" id="menu-v8">
                    <li class="nav-item" role="presentation">
                      <router-link :to="{name: 'Portada'}">Portada</router-link> 
                    </li>
                    <li class="nav-item" role="presentation">
                      <router-link :to="{name: 'SobreMi'}">Sobre Mi</router-link> 
                    </li>
                    <li class="nav-item" role="presentation">
                      <router-link :to="{name: 'Contacto'}">Contacto</router-link>  
                    </li>
                    <li class="nav-item" role="presentation">
                      <router-link to="/post/1">Ãšltimo Post</router-link> 
                    </li>
                </ul>
            </div>
        </div>
    </nav>
    <transition name="efectos">
      <router-view></router-view>
    </transition>
  </div>
</template>

<script>
export default {
  name: 'App'
}
</script>

<style scoped>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;  
}
#menu-v8 a{
  text-shadow: 1px 1px #000;
}
.efectos-enter-active,  
.efectos-leave-active {
  transition: opacity .1s;
}
.efectos-enter, 
.efectos-leave-to {
  opacity: 0;

}
</style>
