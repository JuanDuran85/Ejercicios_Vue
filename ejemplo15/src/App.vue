<template>
  <div id="app">
      <img src="https://lyl-ingenieria.com/blog/wp-content/uploads/2017/08/INTEGRA-WAREHOUSE.jpg" alt="almacen">
      <h1 v-text="titulo"></h1>
      <hr>
      <p v-text="ocupado"></p>
      <p v-text="totalVentas"></p>
      <p v-text="totalDeudas"></p>
      <p v-text="creditos"></p>
      <h3>Productos Disponibles en el almacen:</h3>
      <ul>
        <li v-for="(item, index) in productosConCantidad" :key="index">
          <label v-text="item.name"></label>
        </li>
      </ul>
      <h3>Productos Baratos Disponibles en el almacen:</h3>
      <ul>
        <li v-for="(item, index) in productosConCantidadBaratos" :key="index">
          <label v-text="item.name"></label>
        </li>
      </ul>
      <h3>Busqueda de producto por Identificador: </h3>
      <input type="text" v-model="buscarID">
      <ul v-if="productoPorId.length > 0">
        <li v-for="(item, index) in productoPorId" :key="index">
          <label v-text="item.name"></label>
        </li>
      </ul>
      <p v-else>...</p>
      <hr>
      <p>
        <label v-text="ocupado"></label>
        <input type="button" value="Cambiar el estado" @click="modificandoOcupado">
      </p>
      <hr>
      <hr>
      <p>
        <label v-text="creditos"></label>
        <input type="button" value="Cambiar el estado" @click="modificandoCredito">
      </p>
      <hr>
      <h3>Productos</h3>
      <ul>
        <li v-for="(item, index) in todosLosProductos" :key="index">
          <span>{{item.id}} |  {{item.name}} |  {{item.stock}} | {{item.precio}} |</span>
          <button @click="agregandoStock(item.id)">+</button> 
          <button @click="actualizarVentaProducto(item)">Vender</button>
        </li>
      </ul>
  </div>
</template>

<script>
// se llaman de esta forma porque son objetos de vuex
import {mapState, mapGetters, mapMutations, mapActions} from 'vuex';

export default {
  name: 'App',
  data() {
    return {
      buscarID: ""
    }
  },
  methods: {
    ...mapMutations(['modificandoOcupado']),
    ...mapActions(['modificandoCredito']),
    agregandoStock(productoEnviado){
      this.$store.commit('agregandoStock',productoEnviado);
    },
    actualizarVentaProducto(productoVendido){
      this.$store.dispatch('actualizarVentaProducto',productoVendido)
    }
  },
  computed: {
    titulo(){
      return "Almecen Web";
    },
    ...mapGetters([
      'productosConCantidad'
    ]),
    productosConCantidadBaratos(){
      return this.$store.getters.productosConCantidadBaratos;
    },
    productoPorId(){
      return this.$store.getters.productoPorId(this.buscarID);
    },
    ...mapState({
      todosLosProductos(state){
        return state.productos;
      },
      ocupado(){
        // Al proporcionar la opción de tienda a la instancia raíz, la tienda se inyectará en todos los componentes secundarios de la raíz y estará disponible en ellos como this.$store
        let ocupado = this.$store.state.ocupado ? "Ocupado":"Disponible";
        return `Estado: ${ocupado}`;
      },
      creditos(state){
        return state.creditos ? "Tiene Credito":"No tiene credido";
      },
      totalVentas(){
        console.log(this.$store.state.totalVentas);
        return `Total Ventas: $${this.$store.state.totalVentas}`;
      },
      totalDeudas(state){
        return `Total Deudas: $${state.totalDeudas}`;
      }
    }),
  }
};
</script>

<style>
#app {
  font-size: 20px;
  background-color: azure;
}

#app img {
  width: 300px;
}
</style>
