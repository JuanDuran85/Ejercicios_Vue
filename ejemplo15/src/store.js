import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

// generando instancia de Vuex
const store = new Vuex.Store({
    //utilizacion de estados
    state: {
        ocupado: false,
        totalVentas: 0,
        totalDeudas: 345,
        creditos: true,
        productos: [
            {id: '0001', name: "Guitarra Eléctrica I", stock: "24", precio: "830000"},
            {id: '0002', name: "Bajo Eléctrico II", stock: "12", precio: "330000"},
            {id: '0003', name: "Teclado Electrónico III", stock: "0", precio: "530000"},
            {id: '0004', name: "Guitarra Acustica Clasica IV", stock: "10", precio: "164000"},
            {id: '0005', name: "Piano de Cola Completa V", stock: "0", precio: "3430000"},
        ],
    },
    getters: {
        productosConCantidad: state => {
            return state.productos.filter((productos) => {return productos.stock > 0});
        },
        productosConCantidadBaratos: (state,getters) => {
            return getters.productosConCantidad.filter((barato)=>{return barato.precio < 500000})
        },
        productoPorId: (state,getters) => (id) => {
            return getters.productosConCantidad.filter((producto)=>{return producto.id == id});
        },
    },
    mutations: {
        modificandoOcupado: (state) => {
            state.ocupado = !state.ocupado;
        },
        modificandoCredito: (state) => {
            state.creditos = !state.creditos;
        }, 
        agregandoStock: (state,productoRecibido)=> {
            state.productos.map((produc)=>{
                if (produc.id == productoRecibido) {
                    produc.stock++;
                }
            })
        },
        venderProducto: (state,productoVendido) => {
            state.productos.forEach((produc)=>{
                if (produc.id == productoVendido.id) {
                    if (productoVendido.stock > 0) {
                        produc.stock--;
                        state.totalVentas += parseFloat(productoVendido.precio);
                    } else {
                        alert("No existe inventario para vender el producto");
                    }
                }
            })
        },
    },
    actions: {
        modificandoCredito: context => {
            context.commit('modificandoCredito');
        },
        actualizarVentaProducto(context,productoVendido) {
            context.dispatch('ventaConPromesa',productoVendido).then(()=>{
                alert("Venta exitosa")
            }).catch(()=>{
                console.log("Error de venta por inventario")
            })
        },
        ventaConPromesa(context,productoVendido){
            return new Promise((si,no) => {
                let resultado = false;
                if (productoVendido.stock > 0) {
                    context.commit('venderProducto',productoVendido)
                    resultado = true;
                }
                if (resultado) {
                    si();
                }else {
                    no();
                }
                
            })
        }
    }
});

export default store;