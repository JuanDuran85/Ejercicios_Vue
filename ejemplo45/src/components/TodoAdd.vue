<template>
  <div>
    <form @submit.prevent="addTodo">
      <input class="ToDo-input" type="text" placeholder="Agrega una tarea" v-model="tarea">
      <button @click.prevent.stop="addTodo">Agregar</button>
    </form>
  </div>
</template>

<script>
export default {
  name: 'TodoAdd',
  data() {
    return {
      tarea: '',
    }
  },
  methods: { 
    addTodo(){
      if (this.tarea){
        this.$store.dispatch('agregando',this.tarea);
        this.tarea = '';
      } else {
        console.log("Debe agregar una actividad");
      }
    } 
  }
}
</script>

<style scoped>

</style>
