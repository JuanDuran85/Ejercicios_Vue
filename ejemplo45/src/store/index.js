import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    todos: [
      {
        id: 0,
        task: 'Tarea 1',
        completo: false
      },
      {
        id: 1,
        task: 'Tarea 2',
        completo: false
      },
      {
        id: 2,
        task: 'Tarea 3',
        completo: false
      }
    ]
  },
  getters: {
    mostrarTodos: (state)=>{
      return state.todos;
    }
  },
  mutations: {
    agregarTarea: (state,tareaRecibida)=>{
      let idNuevo = Math.floor(Math.random() * 10);
      console.log(idNuevo)
      while (idNuevo){
        let buscar = state.todos.find(todo => todo.id == idNuevo);
        if (buscar == undefined) {
          break;
        }else if (buscar.id == idNuevo) {
          idNuevo++;
          console.log(idNuevo);
        }
      }
      let tarea = {
        id: idNuevo,
        task: tareaRecibida,
        completo: false
      }
      state.todos.unshift(tarea);
    },
    eliminarTarea: (state,id) => {
      let borrar = state.todos.findIndex(todo => todo.id === id)
      state.todos.splice(borrar,1);
    },
    actualizarTarea: (state,id)=> {
      let actualizar = state.todos.find(todo => todo.id === id)
      actualizar.completo = !actualizar.completo;
    }
  },
  actions: {
    agregando: (context,tareaRecibida) => {
      context.commit('agregarTarea',tareaRecibida);
    },
    eliminando: (context,id) => {
      context.commit('eliminarTarea',id);
    },
    actualizando: (context,id) => {
      context.commit('actualizarTarea',id);
    }
  },
});
