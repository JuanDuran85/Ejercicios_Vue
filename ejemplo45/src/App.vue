<template>
  <div id="app" class="ToDo-container">
    <h1 class="ToDo-title">Lista de tareas con Vuex</h1>
    <todo-add></todo-add>
    <todo-list></todo-list>
  </div>
</template>

<script>
import TodoAdd from './components/TodoAdd.vue';
import TodoList from './components/TodoList';

export default {
  name: 'App',
  components: {
    TodoAdd,
    TodoList,
  }
}
</script>

<style>
:root {
  --color-primary: #41B883;
  --color-secondary: #35495E;
}

html {
  font-family: sans-serif;
  font-size: 16px;
  line-height: 1.6;
}

body {
  margin: 0;
  height: 100vh;
  display: flex;
  background: linear-gradient(180deg, var(--color-secondary), var(--color-primary));
  color: #333;
}

a {
  display: block;
  margin: 1rem auto;
  text-align: center;
  color: var(--color-secondary);
  transition: all .3s ease-out;
}

a:hover {
  color: var(--color-primary);
}

.ToDo-container {
  margin: auto;
  padding: 1rem;
  width: 24rem;
  background-color: #FFF;
  border-radius: .5rem;
  box-shadow: 0 0 2rem rgba(0, 0, 0, .25);
}

.ToDo-title {
  text-align: center;
  margin-top: 0;
}

.ToDo-input {
  width: 100%;
  padding: .5rem;
  font-size: 1rem;
  outline: none;
  border-radius: .25rem;
  border: solid 1px #EEE;
  box-sizing: border-box;
}

.ToDo-.tasks {
  padding-left: 1.5rem;
}

.ToDo-task {
  margin-bottom: .5rem;
}

.ToDo-task:hover {
  cursor: pointer;
}

.completed {
  text-decoration: line-through;
  background-color: rgba(148, 41, 41, 0.151);
  color: #555;
}
</style>
