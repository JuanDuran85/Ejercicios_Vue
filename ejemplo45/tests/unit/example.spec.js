import { shallowMount } from '@vue/test-utils';
import TodoAdd from '@/components/TodoAdd.vue';
import App from '@/App.vue'
//import TodoList from '@/components/TodoList';
import store from '@/store';
import Vuex from 'vuex';

describe('store Vuex', () => {
  beforeAll(()=>{
    Vue.use(Vuex)
    store = new Vuex.Store(store)
  })
})

describe('action', ()=> {
  it('prueba action', () => {
   store.dispatch('agregando','tarea ejemplo 1')
   expect(store.state.todos.length).toBe(4)
  })
})

describe('getters',()=>{
  it('prueba getters', () => {
    let todos = store.getters.mostrarTodos;
    expect(todos.length).toBe(4);
  })
})

describe('TodoAdd.vue', ()=> {
  it('prueba test 1', () => {
    const wrapper = shallowMount(TodoAdd);
    const dataTodoAdd = TodoAdd.data();
    expect(dataTodoAdd.tarea).toEqual('');
    let entrada = wrapper.find('.ToDo-input');
    expect(entrada.element.placeholder).toBe('Agrega una tarea');
  })
  
  it('prueba boton', () => {
    const wrapper = shallowMount(TodoAdd);
    expect(wrapper.find('button').text()).toEqual('Agregar');
    //console.log(wrapper.vm.$data.tarea);
    let variable = wrapper.vm.$data.tarea;
    variable = '';
    wrapper.find('button').trigger('click');
    expect(wrapper.vm.$data.tarea).toBe('');
    wrapper.vm.addTodo = jest.fn();
    wrapper.vm.addTodo();
    expect(wrapper.vm.addTodo.mock.calls.length).toBe(1);
  })
})

describe('App.vue', ()=> {
  it('prueba test 2', () => {
    const wrapper = shallowMount(App);
    expect(wrapper.find('.ToDo-title').element.textContent).toBe('Lista de tareas con Vuex');
    expect(wrapper.find('.ToDo-title').text().length).toBe(24);
    expect(wrapper.element).toMatchSnapshot();
  })
})

/* describe('TodoList', () => {
  test('prueba test 4',()=>{
    const wrapper = shallowMount(TodoList);
    let todos = store.getters.mostrarTodos;
    expect(todos).toBe(todos)
    expect(wrapper.html()).toContain('<ul class="ToDo-tasks">')
  })
}) */
