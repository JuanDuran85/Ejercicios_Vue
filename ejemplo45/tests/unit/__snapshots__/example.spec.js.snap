// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue prueba test 2 1`] = `
<div
  class="ToDo-container"
  id="app"
>
  <h1
    class="ToDo-title"
  >
    Lista de tareas con Vuex
  </h1>
   
  <todo-add-stub />
   
  <todo-list-stub />
</div>
`;
